http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/SharedSdkEventLoopGroup.java

```diff
@@ -90,8 +95,35 @@ private ReferenceCountingEventLoopGroup(EventLoopGroup delegate) {
             // Only want to decrement the reference the first time it's closed. Shutdown is idempotent and may be
             // called multiple times.
             if (hasBeenClosed.compareAndSet(false, true)) {
-                decrementReference();
+                return decrementReference();
             }
+            return CompletedCloseFuture.COMPLETED_FUTURE;
```

Comment:
You can use SucceededFuture with a null executor instead of creating your own future implementation.

Reply:
Yeah, I considered using that but was too afraid to pass two nulls to the constructor. I wish they have an overloaded constructor. Will update to use that instead.